frontend/src/routes/manage/Realm/General.tsx

```diff
@@ -37,14 +62,52 @@ type Props = {
 };
 
 export const General: React.FC<Props> = ({ fragRef }) => {
+    const { t } = useTranslation();
+    const realm = useFragment(fragment, fragRef);
+
+    // We do not allow changing the name of the root realm.
+    if (realm.isRoot) {
+        return <p>{t("manage.realm.general.no-rename-root")}</p>;
+    }
+
+    const { nameSource, ...rest } = realm;
+    if (nameSource === null) {
+        return bug("name source is null for non-root realm");
+    }
+
+    return <NameForm realm={{ nameSource, ...rest }} />;
+};
```

Comment:
```suggestion
    const { nameSource } = realm;
    if (nameSource === null) {
        return bug("name source is null for non-root realm");
    }

    return <NameForm realm={realm} />;
};
```

Reply:
Nop. The `if` changes the type of `nameSource` to be non-null. That's why the code is written that way.

Reply:
Oooh, cool! 👀 

Sometimes I think I should review in a proper editor, not in GitHub. 🤔